tensorflow/cc/framework/gradients.cc

```diff
@@ -525,7 +525,8 @@ Status SymbolicGradientBuilder::AddGradients() {
     for (const Edge* e : n->in_edges()) {
       if (e->IsControlEdge()) continue;
       int dx_index = e->dst_input();
-      if (dx_index >= dx.size()) {
+      const int dx_size = dx.size();
+      if (dx_index >= dx_size) {
```

Comment:
We should change type of `dx_index` local variable instead. It is only used in places where its sign does not matter.

Reply:
++, will split and then handle in split pr.